src/gen9_hevc_encoder.c

```diff
@@ -0,0 +1,7361 @@
+    ALLOC_GPE_2D_RESOURCE(res_brc_constant_data_buffer,
+                          "Brc constant data buffer",
+                          width, height, width);
+
+    width = ALIGN((priv_state->frame_width_4x * 4 + 31) >> 4, 64);
+    height = ALIGN((priv_state->frame_height_4x * 4 + 31) >> 5, 4);
+    ALLOC_GPE_2D_RESOURCE(res_brc_mb_qp_buffer,
+                          "Brc mb qp buffer",
+                          width, height, width);
+
+    //HME scaling buffer allocation
+    width = ALIGN(priv_state->downscaled_width_4x_in_mb * 8, 64);
+    height = ALIGN(priv_state->downscaled_height_4x_in_mb * 4, 8);
+    ALLOC_GPE_2D_RESOURCE(res_brc_me_dist_buffer,
+                          "Brc me distortion buffer",
+                          width, height, width);
+
+    if (generic_state->hme_supported) {
+        width = ALIGN(priv_state->downscaled_width_4x_in_mb * 8, 64);
+        height = ALIGN(priv_state->downscaled_height_4x_in_mb * 4 * 10, 8) * 2;
+        ALLOC_GPE_2D_RESOURCE(s4x_memv_distortion_buffer,
+                              "4x MEMV distortion buffer",
+                              width, height, width);
+
+        width = ALIGN(priv_state->downscaled_width_4x_in_mb * 32,64);
+        height = priv_state->downscaled_height_4x_in_mb * 4 * 10;
+        ALLOC_GPE_2D_RESOURCE(s4x_memv_data_buffer,
+                              "4x MEMV data buffer",
+                              width, height, width);
+       
+        if (generic_state->b16xme_supported) {
+            width = ALIGN(priv_state->downscaled_width_16x_in_mb * 32, 64);
+            height = priv_state->downscaled_height_16x_in_mb * 2 * 4 * 10;
+            ALLOC_GPE_2D_RESOURCE(s16x_memv_data_buffer,
+                                  "16x MEMV data buffer",
+                                  width, height, width);
+
+            if (generic_state->b32xme_supported) {
+                width = ALIGN(priv_state->downscaled_width_32x_in_mb * 32, 64);
+                height = priv_state->downscaled_height_32x_in_mb * 2 * 4 * 10;
+                ALLOC_GPE_2D_RESOURCE(s32x_memv_data_buffer,
+                                      "32x MEMV data buffer",
+                                      width, height, width);
+            }
+        }
+    }
+
+    if (priv_state->flatness_check_supported) {
+        width = ALIGN(priv_state->width_in_mb * 4, 64);
+        height = priv_state->downscaled_height_4x_in_mb * 4;
+        ALLOC_GPE_2D_RESOURCE(res_flatness_check_surface,
+                              "Flatness check buffer",
+                              width, height, width);
+    }
+
+    if (priv_ctx->scaled_2x_surface_obj)
+          i965_DestroySurfaces(priv_ctx->ctx, &priv_ctx->scaled_2x_surface_id, 1);
+
+    width = priv_state->frame_width_in_max_lcu >> 1;
+    height = priv_state->frame_height_in_max_lcu >> 1;
+    if (priv_state->bit_depth_luma_minus8) {
+        width = ALIGN(width, 32);
+        height = ALIGN(height, 32);
+    }
+
```

Comment:
No. 2XScaling is a new process in HEVC MBEnc. This surface is used by later MBEnc process